auto-increment.md

```diff
@@ -0,0 +1,211 @@
+
+## Working principles
+
+TiDB implements the `AUTO_INCREMENT` implicit assignment in the following way:
+
+For each auto-increment column, a globally visible key-value pair is used to record the maximum ID that has been assigned. In a distributed environment, communication between nodes has some overhead. Therefore, to avoid the issue of write amplification, each TiDB node applies for a batch of consecutive IDs as cache when assigning IDs, and then applies for the next batch of IDs after the first batched is assigned. TiDB nodes do not have to apply to the storage node for IDs at each assigning. For example:
+
+```sql
+create table t(id int unique key AUTO_INCREMENT, c int);
+```
+
+Assume two TiDB instances, `A` and `B`, in the cluster. If you execute an `INSERT` statement for the `t` table on `A` and `B` respectively:
+
+```sql
+insert into t (c) values (1)
+```
+
+Instance `A` might cache the auto-increment ID of `[1,30000]`, and instance `B` might cache the auto-increment ID of `[30001,60000]`. In the future `INSERT` statements, these cached ID of each instance will be assigned to the `AUTO_INCREMENT` column as the default value.
+
+## Basic Features
+
+### Uniqueness
+
+> **Warning:**
+>
+> When the cluster has multiple TiDB instances, if the table schema contains the auto-increment ID, it is recommended not to use explicit insert and implicit assignment at the same time (i.e. using the default value of the auto-increment column and the custom value). Otherwise, it might break the uniqueness of implicitly assigned values.
+
+In the example above, perform the following operations in order:
+
+1. The client inserts a statement `insert into t values (2, 1)` to instance `B`, which sets `id` to `2`. The statement is successfully executed.
+
+2. The client sends a statement `insert into t (c) (1)` to instance `A`. This statement does not specify the value of `id`, so the ID is assigned by `A`. At present, because `A` caches the ID of `[1, 30000]`, it might assign `2` as the value of the auto-increment ID, and increases the local counter by `1`. At this time, the data whose ID is `2` already exists in the database, so the `Duplicated Error` error is returned.
+
+### Increment
+
+TiDB guarantees that implicitly assigned value of the `AUTO_INCREMENT` column are incremental only in the cluster with a single TiDB instance. That is, for the same auto-increment column, the value assigned earlier is smaller than the value assigned later. However, in a cluster with multiple instances, TiDB cannot guarantee that the auto-increment column is incremental.
+
+In the example above, if you first execute an `INSERT` statement in instance `B`, and then execute an `INSERT` statement in instance `A`. Because of the principle of caching auto-increment ID, the auto-increment column might implicitly assign `30002` and `2` respectively. The assigned values are not incremental in the time order.
```

Comment:
```suggestion
In the example above, if you first execute an `INSERT` statement on instance `B`, and then execute an `INSERT` statement on instance `A`. Because of the behavior of caching auto-increment ID, the auto-increment column might be implicitly assigned `30002` and `2` respectively. The assigned values are not incremental in the time order.
```